odl/space/pspace.py

```diff
@@ -1064,6 +1123,158 @@ def ufuncs(self):
         """
         return ProductSpaceUfuncs(self)
 
+    @property
+    def real(self):
+        """Real part of the element.
+
+        Examples
+        --------
+        >>> space = odl.ProductSpace(odl.cn(3), odl.cn(2))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j], [-1+2j, -2-3j]])
+        >>> x_real = x.real
+        >>> real_space = odl.ProductSpace(odl.rn(3), odl.rn(2))
+        >>> expected_result = real_space.element([[1, 2, 3], [-1, -2]])
+        >>> x_real == expected_result
+        True
+        """
+        real_part = [part.real for part in self.parts]
+        return self.space.real_space.element(real_part)
+
+    @real.setter
+    def real(self, newreal):
+        """Setter for the real part.
+
+        This method is invoked by ``x.real = other``.
+
+        Parameters
+        ----------
+        newreal : array-like or scalar
+            Values to be assigned to the real part of this element.
+
+        Examples
+        --------
+        >>> space = odl.ProductSpace(odl.cn(3), odl.cn(2))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j], [1 + 1j, 2]])
+        >>> zero = space.real_space.zero()
+        >>> x.real = zero
+        >>> x
+        space.element([[1j, 0j, -3j], [1j, 0]])
```

Comment:
You are right... but it seems like doc-tests are not run for setters because it does not fail. And neither does
```python
>>> False
True
```
I will try to have a look at it

Reply:
This is apparently a "feature" and all the documentation should go into the getter

https://stackoverflow.com/questions/23341106/python-2-7-doctests-ignored-in-setter-method-of-a-class

Reply:
I'll make a general issue for this since we seem to have this error in many places.

Reply:
So apparently it seems like the doc-strings of `.setter` is ignored, see [here](https://stackoverflow.com/questions/23341106/python-2-7-doctests-ignored-in-setter-method-of-a-class). I will simply put the corresponding doc-test in the property itself, while keeping the rest of the documentation in the setter